rs/nervous_system/integration_tests/tests/sns_topics.rs

```diff
@@ -0,0 +1,196 @@
+            &pocket_ic,
+            create_service_nervous_system,
+            sns_instance_label,
+        )
+        .await;
+
+        sns::swap::await_swap_lifecycle(&pocket_ic, sns.swap.canister_id, Lifecycle::Open)
+            .await
+            .unwrap();
+
+        sns
+    };
+
+    // Get an ID of an SNS neuron that can submit proposals. We rely on the fact that this
+    // neuron either holds the majority of the voting power or the follow graph is set up
+    // s.t. when this neuron submits a proposal, that proposal gets through without the need
+    // for any voting.
+    let (sns_neuron_id, sender) = sns::governance::find_neuron_with_majority_voting_power(
+        &pocket_ic,
+        sns.governance.canister_id,
+    )
+    .await
+    .expect("cannot find SNS neuron with dissolve delay over 6 months.");
+
+    let pocket_ic_agent = PocketIcAgent {
+        pocket_ic: &pocket_ic,
+        sender: sender.into(),
+    };
+
+    let governance_canister = GovernanceCanister {
+        canister_id: sns.governance.canister_id,
+    };
+
+    let initial_generic_function = GenericNervousSystemFunction {
+        target_canister_id: Some(PrincipalId::new_user_test_id(42)),
+        target_method_name: Some("do_things".to_string()),
+        validator_canister_id: Some(PrincipalId::new_user_test_id(43)),
+        validator_method_name: Some("validate_things".to_string()),
+        topic: Some(Topic::DaoCommunitySettings),
+    };
+
+    let expected_generic_function = GenericNervousSystemFunction {
+        topic: Some(Topic::ApplicationBusinessLogic),
+        ..initial_generic_function.clone()
+    };
+
+    let initial_function = NervousSystemFunction {
+        id: 1111,
+        name: "Test Custom Proposal Type".to_string(),
+        description: Some("This is a custom proposal type for testing.".to_string()),
+        function_type: Some(FunctionType::GenericNervousSystemFunction(
+            initial_generic_function,
+        )),
+    };
+
+    let expected_function = NervousSystemFunction {
+        function_type: Some(FunctionType::GenericNervousSystemFunction(
+            expected_generic_function,
+        )),
+        ..initial_function.clone()
+    };
+
+    // AddGenericNervousSystemFunction
+    {
+        let response = governance_canister
+            .submit_proposal(
+                &pocket_ic_agent,
+                sns_neuron_id.clone(),
+                Proposal {
+                    title: "Add custom proposal under the DaoCommunitySettings topic.".to_string(),
+                    summary: "Add custom proposal under the DaoCommunitySettings topic."
+                        .to_string(),
+                    url: DUMMY_URL_FOR_PROPOSALS.to_string(),
+                    action: Some(Action::AddGenericNervousSystemFunction(initial_function)),
+                },
+            )
+            .await
+            .unwrap();
+
+        let proposal_id = SubmittedProposal::try_from(response)
```

Comment:
I'm a bit confused why conversion is needed...

Reply:
It's just a way to obtain a substructure from the response. We used in in other tests, so I don't see a reason not to use it here. Note that it's not _necessary_, it just makes the code slightly more concise. Let me know if you disagree.